Jenkinsfile

```diff
@@ -56,118 +95,105 @@ def stepClone()
     }
 }
 
-def stepPrintEnv(board, test)
+def stepRunNodeTests()
 {
-    sh 'dist/tools/ci/print_environment.sh'
+    catchError(buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
+        def tests = []
+        stage( "${env.BOARD} setup"){
+            stepPrepareNodeWorkingDir()
+            tests = stepDiscoverTests()
+        }
+        for (int i=0; i < tests.size(); i++) {
+            stage("${tests[i]}") {
+                Exception caughtException = null
+                catchError(buildResult: 'UNSTABLE', stageResult: 'FAILURE') {
+                    try {
+                        stepPrintEnv()
+                        stepReset(tests[i])
+                        stepMake(tests[i])
+                        stepFlash(tests[i])
+                        stepTest(tests[i])
+                        stepArchiveTestResults(tests[i])
+                    } catch (org.jenkinsci.plugins.workflow.steps.FlowInterruptedException e) {
```

Comment:
why this particular exception?

Reply:
It is the timeout or abort exception.  Without it a timeout will only cancel one test on a node.